runtime/exception.h

```diff
@@ -7,31 +7,60 @@
 #include "runtime/kphp_core.h"
 #include "runtime/memory_usage.h"
 #include "runtime/refcountable_php_classes.h"
+#include "common/algorithms/hashes.h"
+#include "common/wrappers/string_view.h"
 
 array<array<string>> f$debug_backtrace();
 
+struct C$Throwable : public refcountable_polymorphic_php_classes_virt<> {
+  virtual const char *get_class() const noexcept {
+    return "";
+  }
 
-struct C$Exception : refcountable_php_classes<C$Exception> {
-  string message;
-  int64_t code = 0;
-  string file;
-  int64_t line = 0;
-  array<void *> raw_trace;
-  array<array<string>> trace;
+  virtual int32_t get_hash() const noexcept {
+    return static_cast<int32_t>(vk::std_hash(vk::string_view(get_class())));
+  }
 
-  void accept(InstanceMemoryEstimateVisitor &visitor) {
-    visitor("", message);
-    visitor("", file);
+  virtual void accept(InstanceMemoryEstimateVisitor &visitor) {
+    visitor("", $message);
+    visitor("", $file);
     visitor("", raw_trace);
     visitor("", trace);
   }
+
+  C$Throwable() __attribute__((always_inline)) = default;
+  ~C$Throwable() __attribute__((always_inline)) = default;
+
+  string $message;
+  int64_t $code = 0;
+  string $file;
+  int64_t $line = 0;
+  array<void *> raw_trace;
+  array<array<string>> trace;
 };
 
+struct C$Exception : public C$Throwable {
+  static int classname_hash;
```

Comment:
I removed it. If @eugene536 knows a way to get our classname hash function to be a constexpr, we could have a nice thing like:
```cpp
  int32_t get_hash() const noexcept override {
    return constexpr_class_hash("ClassName");
  }
```
For our code, this optimization is not essential, as there should be 0 catch blocks that need to do a class checking, but if anyone would actually try to use this feature for real, it would be unfortunate to compute a hash for every catch clause.